evalml/pipelines/components/transformers/encoders/ordinal_encoder.py

```diff
@@ -0,0 +1,289 @@
+
+from evalml.pipelines.components import ComponentBaseMeta
+from evalml.pipelines.components.transformers.transformer import Transformer
+from evalml.utils import infer_feature_types
+
+"""A transformer that encodes ordinal features."""
+
+
+class OrdinalEncoderMeta(ComponentBaseMeta):
+    """A version of the ComponentBaseMeta class which includes validation on an additional ordinal-encoder-specific method `categories`."""
+
+    METHODS_TO_CHECK = ComponentBaseMeta.METHODS_TO_CHECK + [
+        "categories",
+        "get_feature_names",
+    ]
+
+
+class OrdinalEncoder(Transformer, metaclass=OrdinalEncoderMeta):
+    """A transformer that encodes ordinal features as an array of ordinal integers representing the relative order of categories.
+
+    Args:
+        features_to_encode (list[str]): List of columns to encode. All other columns will remain untouched.
+            If None, all appropriate columns will be encoded. Defaults to None. The order of columns does not matter.
+        categories (list[list[str]]): A two dimensional list of categories, where `categories[i]` is a list of the categories
+            for the column at index `i` in the dataframes passed in at fit and transform.
+            The order of categories specified for a column does not matter.
+            Any category not present in categories will be handled as an unknown value.
+            To not have unknown values raise an error, set handle_unknown to "use_encoded_value".
+            Defaults to None.
+        handle_unknown ("error" or "use_encoded_value"): Whether to ignore or error for unknown categories
+            for a feature encountered during `fit` or `transform`. When set to "error",
+            an error will be raised when an unknown category is found.
+            When set to "use_encoded_value", unknown categories will be encoded as the value given
+            for the parameter unknown_value. Defaults to "error."
+        unknown_value (int or np.nan): The value to use for unknown categories seen during fit or transform.
+            Required when the parameter handle_unknown is set to "use_encoded_value."
+            The value has to be distinct from the values used to encode any of the categories in fit.
+            Defaults to None.
+        encoded_missing_value (int or np.nan): The value to use for missing (null) values seen during
+            fit or transform. Defaults to np.nan.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+        # --> is random seed still necessary?
+    """
+
+    name = "Ordinal Encoder"
+    hyperparameter_ranges = {}
+    """{}"""
+
+    def __init__(
+        self,
+        features_to_encode=None,
+        categories=None,
+        handle_unknown="error",
+        unknown_value=None,
+        encoded_missing_value=None,
+        random_seed=0,
+        **kwargs,  # --> why do we have kwargs if they arent used?? Maybe needed for larger evalml pipeline
```

Comment:
To answer your question here, kwargs is generally added across all components in case the user wants to set parameters in the component object that we don't include in our parameters. 

Nominally, with the ordinal encoder the sklearn implementation offers a `dtype` parameter that we don't expose, but the user could choose to set here. Since we don't instantiate the component until fit time this becomes a little moot. I'm not sure whether we should worry about passing them into the component object or not, but we should at least keep this function signature the same. The same holds for the `random_seed` parameter which I see you've commented on as well.

Reply:
Awesome--I'll keep them in. Thanks for explaining!